src/test/scala/io/github/spark_redshift_community/spark/redshift/TestUtils.scala

```diff
@@ -94,6 +94,29 @@ object TestUtils {
     calendar.getTime.getTime
   }
 
+  def toNanosTimestamp(
+    year: Int,
+    zeroBasedMonth: Int,
+    date: Int,
+    hour: Int,
+    minutes: Int,
+    seconds: Int,
+    nanos: Int
+                               ): Timestamp = {
+    val ts = new Timestamp(
+      toMillis(
+        year,
+        zeroBasedMonth,
+        date,
+        hour,
+        minutes,
+        seconds
+      )
+    )
+    ts.setNanos(nanos)
+    ts
+  }
+
```

Comment:
Redshift doesn't support anything past microseconds, so I'd say no reason to test nanos.

Reply:
```
> select '2019-01-01 00:00:00.000001' :: timestamp           -- microsecond                                                                                                                                                               
+----------------------------+
| timestamp                  |
|----------------------------|
| 2019-01-01 00:00:00.000001 |
+----------------------------+

> select '2019-01-01 00:00:00.0000001' :: timestamp          -- tenth of a microsecond gets truncated                                                                                                                                                 
+---------------------+
| timestamp           |
|---------------------|
| 2019-01-01 00:00:00 |
+---------------------+
```

Reply:
I am not testing nanos here.
The Timestamp object has only a `.setNanos()` method and I can only use that to set microseconds. If you check the test, I am doing `nanos=123456000` that is equivalent to have 123456 microseconds